### PR TITLE
fix: configuration subscribe nil ptr map

### PR DESCRIPTION
**Pull request by LaurenceLiZhixin**
# Description

## Issue reference

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [x] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [x] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [x] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/grpc/api.go [hunk 1]
```diff
@@ -144,6 +144,7 @@ func NewAPI(
 		transactionalStateStores: transactionalStateStores,
 		secretStores:             secretStores,
 		configurationStores:      configurationStores,
+		configurationSubscribe:   make(map[string]bool, 0),
```
Comment (daixiang0) [inline pkg/grpc/api.go:147]: Please fix lint issue.

Comment (LaurenceLiZhixin) [reply pkg/grpc/api.go]: done

File: pkg/grpc/api.go [hunk 2]
```diff
@@ -1259,10 +1260,6 @@ type configurationEventHandler struct {
 }
 
 func (h *configurationEventHandler) updateEventHandler(ctx context.Context, e *configuration.UpdateEvent) error {
-	if h.api.appChannel == nil {
```
Comment (mukundansundar) [inline pkg/grpc/api.go:1262]: why's this check removed?

**Comment by yaron2** [comment]
This is already fixed in 1.5 via https://github.com/dapr/dapr/pull/3971

**Pull request closed without merging**